pkg/autodiscover/autodiscover_test.go

```diff
@@ -15,3 +15,171 @@
+		expectedOutput string
+	}{
+		{
+			testPrefix:     "pre1",
+			testLabelName:  "label1",
+			expectedOutput: "pre1/label1",
+		},
+		{
+			testPrefix:     "",
+			testLabelName:  "label1",
+			expectedOutput: "label1",
+		},
+		{
+			testPrefix:     "pre1",
+			testLabelName:  "",
+			expectedOutput: "pre1/",
+		},
+		{
+			testPrefix:     "",
+			testLabelName:  "",
+			expectedOutput: "",
+		},
+	}
+
+	for _, tc := range testCases {
+		assert.Equal(t, tc.expectedOutput, buildLabelName(tc.testPrefix, tc.testLabelName))
+	}
+}
+
+func TestBuildLabelQuery(t *testing.T) {
+	testCases := []struct {
+		testLabel      configuration.Label
+		expectedOutput string
+	}{
+		{
+			testLabel: configuration.Label{
+				Prefix: "prefix1",
+				Name:   "name1",
+				Value:  "value1",
+			},
+			expectedOutput: "prefix1/name1=value1",
+		},
+		{
+			testLabel: configuration.Label{
+				Prefix: "prefix2",
+				Name:   "",
+				Value:  "value2",
+			},
+			expectedOutput: "prefix2/=value2",
+		},
+		{
+			testLabel: configuration.Label{
+				Prefix: "",
+				Name:   "name3",
+				Value:  "value3",
+			},
+			expectedOutput: "name3=value3",
+		},
+		{
+			testLabel: configuration.Label{
+				Prefix: "",
+				Name:   "",
+				Value:  "value4",
+			},
+			expectedOutput: "=value4",
+		},
+	}
+
+	for _, tc := range testCases {
+		assert.Equal(t, tc.expectedOutput, buildLabelQuery(tc.testLabel))
+	}
+}
+
+func TestBuildLabelKeyValue(t *testing.T) {
+	testCases := []struct {
+		testLabel                configuration.Label
+		expectedKey, expectedVal string
+	}{
+		{
+			testLabel: configuration.Label{
+				Prefix: "prefix1",
+				Name:   "name1",
+				Value:  "value1",
+			},
+			expectedKey: "prefix1/name1",
+			expectedVal: "value1",
+		},
+		{
+			testLabel: configuration.Label{
+				Prefix: "prefix2",
+				Name:   "",
+				Value:  "value2",
+			},
+			expectedKey: "prefix2/",
+			expectedVal: "value2",
+		},
+		{
+			testLabel: configuration.Label{
+				Prefix: "",
+				Name:   "name3",
+				Value:  "value3",
+			},
+			expectedKey: "name3",
+			expectedVal: "value3",
+		},
+		{
+			testLabel: configuration.Label{
+				Prefix: "",
+				Name:   "",
+				Value:  "value4",
+			},
+			expectedKey: "",
+			expectedVal: "value4",
+		},
+	}
+
+	for _, tc := range testCases {
+		k, v := buildLabelKeyValue(tc.testLabel)
+		assert.Equal(t, tc.expectedKey, k)
+		assert.Equal(t, tc.expectedVal, v)
+	}
+}
+
+func TestNamespacesListToStringList(t *testing.T) {
+	testCases := []struct {
+		testList       []configuration.Namespace
+		expectedOutput []string
+	}{
+		{
+			testList: []configuration.Namespace{
+				{
+					Name: "ns1",
+				},
+				{
+					Name: "ns2",
+				},
+			},
+			expectedOutput: []string{"ns1", "ns2"},
+		},
+		{
+			testList:       []configuration.Namespace{},
+			expectedOutput: nil,
+		},
+		{
+			testList: []configuration.Namespace{
+				{Name: "name1"},
+				{Name: "name1"},
+			},
+			expectedOutput: []string{"name1", "name1"},
```

Comment:
Isn't this test case equivalent to the first one?

Reply:
I was trying to do two different and two same strings for the different testcases.